### PR TITLE
Only show validated TRNs after ECF validation

### PR DESCRIPTION
**Pull request by ethanmills**
There is a chance that something could go wrong at ECF TRN validation
that means a TRN we get from NPQ is not correct. Be conservative in
declaring that the TRN is validated.

Also, share some TRN pre-validation

### CODE REVIEW

**Review by jamesrose — APPROVED** [review]

**Comment by ethanmills** [comment]
I always forget about the other set of specs 🙄

File: app/serializers/participant_serializer.rb [hunk 1]
```diff
@@ -16,6 +16,17 @@ def active_participant_attribute(attr, &blk)
...
+    def trn(user)
+      user.teacher_profile.trn || user.teacher_profile.ecf_profile.ecf_participant_validation_data&.trn
+    end
+
+    def validated_trn(user)
+      eligibility_status = user.teacher_profile.ecf_profile.ecf_participant_eligibility&.status
+      if %w[matched eligible].include?(eligibility_status)
+        user.teacher_profile.trn
+      end
+    end
```
Comment (tonyheadford) [inline app/serializers/participant_serializer.rb:29]: So, if there is an existing TRN from NPQ and we validate a different TRN, we always use the existing TRN not the validated one?

File: spec/serializers/participant_serializer_spec.rb [hunk 2]
```diff
@@ -4,20 +4,30 @@
...
     let(:mentor_cohort) { mentor.mentor_profile.cohort }
 
-    it "outputs correctly formatted serialized Mentors" do
-      expected_json_string = "{\"data\":{\"id\":\"#{mentor.id}\",\"type\":\"participant\",\"attributes\":{\"email\":\"#{mentor.email}\",\"full_name\":\"#{mentor.full_name}\",\"mentor_id\":null,\"school_urn\":\"#{mentor.mentor_profile.school.urn}\",\"participant_type\":\"mentor\",\"cohort\":#{mentor_cohort.start_year},\"status\":\"active\",\"teacher_reference_number\":\"#{mentor.teacher_profile.trn}\",\"teacher_reference_number_validated\":true,\"eligible_for_funding\":null,\"pupil_premium_uplift\":null,\"sparsity_uplift\":null}}}"
-      expect(ParticipantSerializer.new(mentor).serializable_hash.to_json).to eq expected_json_string
-    end
+    context "when the participant is eligible for funding" do
+      before do
+        eligibility = ECFParticipantEligibility.create!(participant_profile: ect_profile)
+        eligibility.update!(status: :eligible)
```
Comment (tonyheadford) [inline spec/serializers/participant_serializer_spec.rb:17]: You're missing lots of sugar, as this is an enum you can do this instead:
```suggestion
        eligibility.eligible_status!
```

File: app/serializers/participant_serializer.rb (hunk 1 above)
Comment (tonyheadford) [reply app/serializers/participant_serializer.rb:29]: Although in that case we would be `:manual_check` won't we?

Comment (ethanmills) [reply app/serializers/participant_serializer.rb:29]: Yep - so we will share the NPQ one and `teacher_reference_number_validated`  will be false

**Review by tonyheadford — APPROVED** [review]

**Pull request merged**